### ISSUE TITLE
ESLint@^8 cannot format the code when saving

### ISSUE THREAD

**Issue by vvizden** [microsoft/vscode-eslint#1547]
Hi,
After my test, ESLint@^8 cannot format the code when saving with the following configuration, and versions ^5 to ^7 do fine.
```json
"editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
}
```
I don't know if this issue is reported to eslint or vscode-eslint.

**Comment by MariaSolOs** [comment]
Try setting `"eslint.format.enable": true`

**Comment by vvizden** [comment]
> Try setting `"eslint.format.enable": true`

Doesn't seem to work.

**Comment by dbaeumer** [comment]
What happens when running ESLint in the terminal on the same file with --fix. Will it be fixed. If yes, can you then please provide me with a GitHub repository I can clone that demos what you are seeing. 

We use ESLint 8 in day-to-day developement as well and format on save works for us (doesn't mean that there is not a bug somewhere, but I need to be able to reproduce it)

**Comment by vvizden** [comment]
> What happens when running ESLint in the terminal on the same file with --fix. Will it be fixed. If yes, can you then please provide me with a GitHub repository I can clone that demos what you are seeing.
> 
> We use ESLint 8 in day-to-day developement as well and format on save works for us (doesn't mean that there is not a bug somewhere, but I need to be able to reproduce it)

Here <https://github.com/vvizden/vscode-eslint8-saving>

**Comment by MariaSolOs** [comment]
@vvizden Your repro doesn't seem to include a `settings.json` file. Is that intentional?

**Comment by vvizden** [comment]
> @vvizden Your repro doesn't seem to include a `settings.json` file. Is that intentional?

I used the VSCode *user settings* and now I submit the settings.json.

**Comment by vvizden** [comment]
@MariaSolOs You reminded me, and I found the reason through the binary comment in my user settings.json file.
Seems because of this config:
```json
"eslint.codeActionsOnSave.rules": [],
```
But why is versions ^5 to ^7 not affected by it.

**Comment by MariaSolOs** [comment]
@vvizden I just cloned your repo and everything worked for me.

> Seems because of this config:
`"eslint.codeActionsOnSave.rules": [],`
That makes sense. That setting lists the rules that should be auto-fixed when saving/formatting a file. If it's empty, the extension will not make any changes. If not defined, then the default is to use the rules in the `.eslintrc` file.

> But why is versions ^5 to ^7 not affected by it.
Good question. The README says:
![image](https://user-images.githubusercontent.com/62502207/200466302-68ed343d-052e-47e0-aa2c-c6bafb561c00.png)
So what I think was happening is that with earlier versions of ESLint this extension would just ignore your `"eslint.codeActionsOnSave.rules": []` setting, but now that you've upgraded to v8 it results in the behavior explained above.

**Comment by vvizden** [comment]
Thank you very much for the careful answer, and here is a summary to help friends who encounter the same problem.

Solve this problem quickly：

It doesn't matter which version of eslint ( now the latest version is 8.27.0 ) in package.json, just remove the `"eslint.codeActionsOnSave.rules": []` in your VSCode settings.json.
